app/src/org/commcare/android/adapters/WiFiDirectAdapter.java

```diff
@@ -0,0 +1,109 @@
+
+import java.util.ArrayList;
+
+/**
+ * Created by willpride on 12/29/15.
+ */
+public class WiFiDirectAdapter extends SquareButtonAdapter {
+
+    private final CommCareWiFiDirectActivity wiFiDirectActivity;
+
+    public WiFiDirectAdapter(CommCareWiFiDirectActivity activity) {
+        super(activity);
+
+        this.wiFiDirectActivity = activity;
+    }
+
+    private static View.OnClickListener getSendButtonListener(final CommCareWiFiDirectActivity activity) {
+        return new View.OnClickListener() {
+            public void onClick(View v) {
+                activity.prepareFileTransfer();
+            }
+        };
+    }
+
+    private static View.OnClickListener getDiscoverButtonListener(final CommCareWiFiDirectActivity activity) {
+        return new View.OnClickListener() {
+            public void onClick(View v) {
+                activity.discoverPeers();
+            }
+        };
+    }
+
+    private static View.OnClickListener getSubmitButtonListener(final CommCareWiFiDirectActivity activity) {
+        return new View.OnClickListener() {
+            public void onClick(View v) {
+                activity.submitFiles();
+            }
+        };
+    }
+
+    private static View.OnClickListener getChangeModeButtonListener(final CommCareWiFiDirectActivity activity) {
+        return new View.OnClickListener() {
+            public void onClick(View v) {
+                activity.changeState();
+            }
+        };
+    }
+
+    @Override
+    protected HomeCardDisplayData getItem(int position) {
+        return getButtonDisplayData().get(position);
+    }
+
+    private ArrayList<HomeCardDisplayData> getButtonDisplayData() {
+        ArrayList<HomeCardDisplayData> buttonData = new ArrayList<HomeCardDisplayData>();
+
+        HomeCardDisplayData sendButton = HomeCardDisplayData.homeCardDataWithStaticText("Send",
+                R.color.white,
+                R.drawable.wifi_direct_transfer,
+                R.color.cc_attention_positive_color,
+                getSendButtonListener(wiFiDirectActivity));
+        HomeCardDisplayData discoverButton = HomeCardDisplayData.homeCardDataWithStaticText("Discover",
+                R.color.white,
+                R.drawable.wifi_direct_discover,
+                R.color.cc_light_cool_accent_color,
+                getDiscoverButtonListener(wiFiDirectActivity));
+        HomeCardDisplayData submitButton = HomeCardDisplayData.homeCardDataWithStaticText("Submit",
+                R.color.white,
+                R.drawable.wifi_direct_submit,
+                R.color.solid_dark_orange,
+                getSubmitButtonListener(wiFiDirectActivity));
+        HomeCardDisplayData changeModeButton = HomeCardDisplayData.homeCardDataWithStaticText("Change Mode",
+                R.color.white,
+                R.drawable.wifi_direct_change_mode,
+                R.color.cc_brand_text,
+                getChangeModeButtonListener(wiFiDirectActivity));
```

Comment:
Ideally we wouldn't create new `HomeCardDisplayData` objects every time `getItem` is called. A list of them can be stored statically and filtered every time `getButtonDisplayData` is called. There are probably even better ways of doing it using some sort of listener for changes to the `wiFiDirectActivity.mState`, but those might be overly complex.